iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -584,6 +586,7 @@ private async Task HandleIncomingMessagesAsync()
             // This is a user-supplied callback that isn't required to be awaited by us. We can simply invoke it and continue.
             _ = _deviceMessageReceivedListener?.Invoke(message);
             await TaskHelpers.CompletedTask.ConfigureAwait(false);
+            await CompleteIncomingMessageAsync(message).ConfigureAwait(false);
```

Comment:
Since we're now awaiting a task inside this method, you can remove the `await TaskHelpers.CompletedTask.ConfigureAwait(false);` call. 
That was previously added since this async method expected a Task in return but we weren't awaiting any tasks in here.